plugins/provisioners/chef/provisioner/chef_zero.rb

```diff
@@ -101,6 +102,12 @@ def verify_shared_folders(folders)
           end
         end
 
+        def verify_chef_nodes_folder
+          if !File.exists? @config.nodes_path[0][1]
+            raise ChefError, :missing_chef_node_folder
+          end
+        end
```

Comment:
OK, I think instead of doing this, you should instead add a new validation under `chef_zero`'s config validate that ensures each element in the `nodes_path` array exists on disk, and if not, appends a new error message to that `errors` array (make sure it works for the case where it's either just a `string` or `array`, since it looks like it can be both according to the docs). Similar to how it works when the config setting is missing completely. The class is [here](https://github.com/mitchellh/vagrant/blob/master/plugins/provisioners/chef/config/chef_zero.rb#L80-L82). I'd put that method under the highlighted if-statement there.

Also PS don't forget to [add a new test](https://github.com/mitchellh/vagrant/blob/master/test/unit/plugins/provisioners/chef/config/chef_zero_test.rb#L105-L110)! :)